silx/gui/plot/tools/roi.py

```diff
@@ -132,7 +146,7 @@ def getSupportedRoiClasses(cls):
 
     # Associated QActions
 
-    def getInteractionModeAction(self, roiClass):
+    def getInteractionModeAction(self, roiClass, parent=None):
```

Comment:
Why do you need to set the parent of the QActions? It can be as risky as using `self` since they can be shared among multiple widgets.

Reply:
This function is in fact an action factory, the parent should be provided. It least you should be able to chose. When i create an action i want it to be well parented (to a menu or whatever).

I don't know how Qt deal with actions, i guess it can trig destroy to update the menus, but right now I did not find any problems to do so.

Reply:
Yes... but the `QAction` are cached and share a `QActionGroup` which parent is the manager.
As it is now, it is not a factory.
It was initially done with `QDockWidget.toggleViewAction` in mind.

Reply:
I just checked for `QDockWidget.toggleViewAction`: it always returns the same `QAction` and its parent is the `QDockWidget`.